content/rosa/entra-external-auth/index.md

```diff
@@ -0,0 +1,455 @@
+    namespace: default
+    user: oidc
+  name: admin
+current-context: admin
+kind: Config
+preferences: {}
+users:
+- name: oidc
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1
+      args:
+      - oidc-login
+      - get-token
+      - --oidc-issuer-url=https://login.microsoftonline.com/${TENANT_ID}/v2.0
+      - --oidc-client-id=${CLIENT_ID}
+      - --oidc-client-secret=${CLIENT_SECRET}
+      - --oidc-extra-scope=email
+      - --oidc-extra-scope=openid
+      command: kubectl
+      env: null
+      interactiveMode: Never
+```
+Set the KUBECONFIG environment variable to the location of the rosa-cluster.kubeconfig file which will enable the OpenShift CLI to authenticate against the ROSA cluster with the contents of the file.
+
+To confirm that the identity details from the OIDC token is being used by the OpenShift CLI:
+
+```bash
+export KUBECONFIG=$(pwd)/rosa-auth.kubeconfig
+```
+
+Confirm access to cluster with elevated priviledges
+
+```bash
+oc get nodes
+```
+example output:
+
+```bash
+NAME                         STATUS   ROLES    AGE     VERSION
+ip-10-0-0-170.ec2.internal   Ready    worker   3h29m   v1.30.7
+ip-10-0-1-171.ec2.internal   Ready    worker   3h30m   v1.30.7
+ip-10-0-2-161.ec2.internal   Ready    worker   3h29m   v1.30.7
+```
+To verify logged in as user of group run
+
+```bash
+oc auth whoami
+```
+### Validate access to the cluster's API using device-code authentication 
+
+Note: Using device-code authentication is when you are authenticating from somewhere that doesn't have a web browser. The --token-cache-storage=disk parameter is used when a client machine lacks keyring. 
+
+Click on the "Authentication" on your Entra ID and Enable Public flows in Advanced section of the page
+
+![Azure Portal - Enable Public flows Authentication Page](./images/azure-portal-allow-public-flows.png)
+
+Create a **rosa-auth-headless.kubeconfig** file containing Entra ID authentication details and cluster API.
+
+```bash
+apiVersion: v1
+clusters:
+- cluster:
+    server: ${API_URL}
+  name: cluster
+contexts:
+- context:
+    cluster: cluster
+    namespace: default
+    user: oidc
+  name: admin
+current-context: admin
+kind: Config
+preferences: {}
+users:
+- name: oidc
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1
+      args:
+      - oidc-login
+      - get-token
+      - --oidc-issuer-url=https://login.microsoftonline.com/${TENANT_ID}/v2.0
+      - --oidc-client-id=${CLIENT_ID}
+      - --oidc-client-secret=${CLIENT_SECRET}
+      - --oidc-extra-scope=email
+      - --oidc-extra-scope=openid
+      - --grant-type=device-code
+      - --skip-open-browser
+      - --token-cache-storage=disk
+      command: kubectl
+      env: null
+      interactiveMode: Never
+```
+Set the `KUBECONFIG` environment variable to the location of the `rosa-cluster.kubeconfig` file which will enable the OpenShift CLI to authenticate against the ROSA cluster with the contents of the file.
```

Comment:
```suggestion
Set the `KUBECONFIG` environment variable to the location of the `rosa-cluster.kubeconfig` file. This will configure the OpenShift CLI to authenticate against the ROSA cluster with the OIDC client.
```